sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -150,28 +152,39 @@ public DirectoryAsyncClient getSubDirectoryClient(String subDirectoryName) {
      *
      * <p>Create the directory</p>
      *
-     * {@codesnippet com.azure.storage.file.directoryAsyncClient.createWithResponse#Map}
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.createWithResponse#filesmbproperties-string-map}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory">Azure Docs</a>.</p>
      *
+     * @param smbProperties The SMB properties of the directory.
+     * @param filePermission The file permission of the directory.
      * @param metadata Optional metadata to associate with the directory
      * @return A response containing the directory info and the status of creating the directory.
      * @throws StorageException If the directory has already existed, the parent directory does not exist or directory name is an invalid resource name.
      */
-    public Mono<Response<DirectoryInfo>> createWithResponse(Map<String, String> metadata) {
-        return withContext(context -> createWithResponse(metadata, context));
-    }
-
-    Mono<Response<DirectoryInfo>> createWithResponse(Map<String, String> metadata, Context context) {
-        // TODO (alzimmer): These properties are dummy defaults to allow the new service version to be used. Remove these and use correct defaults when known (https://github.com/Azure/azure-sdk-for-java/issues/5039)
-        String fileAttributes = "None";
-        String filePermission = "inherit";
-        String fileCreationTime = "now";
-        String fileLastWriteTime = "now";
-        return postProcessResponse(azureFileStorageClient.directorys().
-            createWithRestResponseAsync(shareName, directoryPath, fileAttributes, fileCreationTime, fileLastWriteTime,
-                null, metadata, filePermission, null, context))
+    public Mono<Response<DirectoryInfo>> createWithResponse(FileSmbProperties smbProperties, String filePermission,
+        Map<String, String> metadata) {
+        return withContext(context -> createWithResponse(smbProperties, filePermission, metadata, context));
+    }
+
+    Mono<Response<DirectoryInfo>> createWithResponse(FileSmbProperties smbProperties, String filePermission,
+        Map<String, String> metadata, Context context) {
+        FileSmbProperties properties = smbProperties == null ? new FileSmbProperties() : smbProperties;
+
+        // Checks that file permission and file permission key are valid
+        FileExtensions.filePermissionAndKeyHelper(filePermission, properties.filePermissionKey());
+
+        // If file permission and file permission key are both not set then set default value
+        filePermission = properties.filePermission(filePermission, FileConstants.FILE_PERMISSION_INHERIT);
```

Comment:
It doesn't look like you check whether permission and key are both not set? Unless that happens in filePermission().

Should we also be accepting the values you're setting for the ntfs stuff instead of setting constants? (Sorry idk much about this feature. I'll trust what you say)

Reply:
I created package private methods in FileSmbproperties. that method filePermission(String, String) checks the internal filePermissionKey and the permission passed in for both null

Reply:
The NTFS stuff is like properties essentially the service will expect a string like this Directory|ReadOnly|Something, so I have an NTFSAttributesEnum that the user can use and we hide the conversion to the expected String if that makes sense. The service also returns the ntfsattributes the same way so I also have a method that converts it back into the enumset